core/src/test/scala/eu/shiftforward/apso/akka/http/ProxySupportSpec.scala

```diff
@@ -153,6 +164,26 @@ class ProxySupportSpec(implicit ee: ExecutionEnv) extends Specification with Spe
         status == OK
         responseAs[String] must be_==("/remote-proxy")
       }
+
+      def parseResult(result: RouteResult): Future[String] = result match {
+        case Complete(res) if res.status.intValue() == 200 =>
+          res.entity.toStrict(10.seconds).map { r => r.data.utf8String }
+        case Complete(res) => Future.successful(res.status.intValue().toString)
+        case _ => Future.failed(new Exception("Failed to parse result"))
+      }
+
+      proxy.sendRequest(Get("/proxied"), failOnDrop = false)
+        .flatMap(parseResult) must be_==("/proxied").awaitFor(10.seconds)
+
+      val badProxy = new Proxy(interface, port, reqQueueSize = 1)
+      // Fill the bad proxy
+      (0 to 10).foreach(x => badProxy.sendRequest(Delete("/proxied-" + x), failOnDrop = false))
```

Comment:
Shouldn't one be enough in this case?

Reply:
Nope. I think that the internal Buffer stores some extra space, but it's hard for me to tell.

`(0 to 4)` seems to do the trick, but I would rather add some extra messages just in case.